motoman_msgs/srv/SelectTool.srv

```diff
@@ -0,0 +1,92 @@
+#
+# NOTE 2: any tool switch initiated by invoking this service WILL however be
+#         taken into account by safety systems such as the FSU and PFL while
+#         executing (Moto)ROS-initiated motions.
+#
+# NOTE 3: active tool will be switched AFTER the controller has completed
+#         execution of the last trajectory segment that was sent to the
+#         controller BEFORE this service was called.
+#
+#         The motion queue internal to MotoROS caches a number of segments in
+#         a FIFO. Only segments received AFTER this service was invoked will
+#         be executed with the new tool. Any segments received before a tool
+#         switch was invoked will use the old tool.
+#
+#         To guarantee motion will be executed using the new tool, applications
+#         should check the 'in_motion' field (part of the RobotStatus message
+#         published on the 'robot_status' topic by the driver) and invoke the
+#         service when the robot has come to a stop (and the motion queue of
+#         MotoROS is empty). Any subsequent trajectories will use the new tool.
+#
+# NOTE 4: as MotoROS currently only executes joint space trajectories, changing
+#         tool file with this service DOES NOT affect TCP poses while executing
+#         trajectories.
+#
+#         To clarify: the TCP definition part of the tool file is NOT taken
+#         into account when calculating manipulator motion based on incoming
+#         ROS JointTrajectoryAction goals (as JointTrajectory goals do not
+#         include any Cartesian poses, only absolute joint space coordinates
+#         for each axis).
+#
+#         Instead, ROS applications should use different TF frames to define
+#         tool frames on the ROS side and plan their motions with the
+#         appropriate TF frame as the active tool.
+#
+#         This service could then be used to notify the controller of other
+#         tool characteristics, such as weight, CoG and inertia by switching
+#         to a tool file configured with those parameters.
+#
+# NOTE 5: there is currently no way to retrieve the active tool file.
+#
+# Refer also the general Yaskawa Motoman documentation on tool file
+# configuration, selection and use.
+
+
+# Numeric ID of the group the tool file is defined for.
+#
+# NOTE: this MUST correspond to a group ID currently defined and active on the
+#       controller. The ROS service does not check whether the value specified
+#       here is legal. The MotoROS server will however check this, will refuse
+#       to switch to an illegal value and will report this to the ROS driver.
+#       Callers will be notified by 'success' set to 'false' and an appropriate
+#       error message being provided via the 'message' field of the service
+#       response.
+#
+# legal-values: [0, total_nr_of_groups)
+# required: yes (absence-causes-service-failure)
+# default: no-default
+int32 group_number
```

Comment:
I've kept these fields as signed integers, as that's what MotoROS has here:

https://github.com/ros-industrial/motoman/blob/1509f9b8cf003a1eeda7161b4a14c95b51b5297e/motoman_driver/MotoPlus/SimpleMessage.h#L271-L276

We could opt to spec `uint32` here to prevent users from setting `< 0` here, at least on the ROS API side of things.

Reply:
There is range checking in the code. But yes... those really should be unsigned.

Reply:
I'll change the fields in the ROS `.srv` to unsigned. One source of potential problems less.